src/lazy/expanded/mod.rs

```diff
@@ -530,8 +532,8 @@ impl<Encoding: Decoder, Input: IonInput> ExpandingReader<Encoding, Input> {
         // See if the raw reader can get another expression from the input stream. It's possible
         // to find an expression that yields no values (for example: `(:void)`), so we perform this
         // step in a loop until we get a value or end-of-stream.
-        let allocator: &BumpAllocator = self.context().allocator();
-        let context_ref = EncodingContextRef::new(allocator.alloc_with(|| self.context()));
+        let context_ref = self.context();
+
```

Comment:
🗺️ This was copying the `EncodingContextRef` to the bump allocator so it would survive the lifetime `'top`...but it was already going to survive that because the `EncodingContext` lives in the `ExpandingReader`.